apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromLocal/VideoFromLocal.tsx

```diff
@@ -1,107 +1,55 @@
-      id
-      image
-      snippet {
-        primary
-        value
-      }
-      title {
-        primary
-        value
-      }
-      variant(languageId: "529") {
-        id
-        duration
-      }
-    }
-  }
-`
-
 interface VideoFromLocalProps {
   onSelect: (block: VideoBlockUpdateInput) => void
 }
 
 export function VideoFromLocal({
   onSelect
 }: VideoFromLocalProps): ReactElement {
-  const [searchQuery, setSearchQuery] = useState<string>('')
-  const [videos, setVideos] = useState<VideoListProps['videos']>()
-  const { loading, fetchMore } = useQuery<GetVideos>(GET_VIDEOS, {
-    variables: {
-      offset: 0,
-      limit: 5,
-      where: {
-        availableVariantLanguageIds: ['529'],
-        title: searchQuery === '' ? null : searchQuery,
-        labels: [
-          VideoLabel.episode,
-          VideoLabel.featureFilm,
-          VideoLabel.segment,
-          VideoLabel.shortFilm
-        ]
-      }
-    },
-    onCompleted: (data) => {
-      setVideos(
-        data.videos.map((video) => ({
-          id: video.id,
-          title: video.title.find(({ primary }) => primary)?.value,
-          description: video.snippet.find(({ primary }) => primary)?.value,
-          image: video.image ?? '',
-          duration: video.variant?.duration,
-          source: VideoBlockSource.internal
-        }))
-      )
-    }
-  })
-  const [hasMore, setHasMore] = useState(true)
   const { t } = useTranslation('apps-journeys-admin')
-  const handleFetchMore = async (): Promise<void> => {
-    const response = await fetchMore({
-      variables: {
-        offset: videos?.length ?? 0
-      }
-    })
-    if (response.data?.videos?.length === 0) {
-      setHasMore(false)
-    } else {
-      setVideos((prevVideos) => [
-        ...(prevVideos ?? []),
-        ...response.data.videos.map((video) => ({
-          id: video.id,
-          title: video.title.find(({ primary }) => primary)?.value,
-          description: video.snippet.find(({ primary }) => primary)?.value,
-          image: video.image ?? '',
-          duration: video.variant?.duration,
-          source: VideoBlockSource.internal
-        }))
-      ])
-    }
+
+  const [searchQuery, setSearchQuery] = useState<string>('')
+
+  const { status } = useInstantSearch()
+  const { hits, showMore, isLastPage } = useInfiniteHits({})
+
+  function transformItems(items): VideoListProps['videos'] {
+    return items.map((videoVariant) => ({
+      id: videoVariant.videoId,
+      title: videoVariant.titles[0],
+      description: videoVariant.description[0],
+      image: videoVariant.image,
+      duration: videoVariant.duration,
+      source: VideoBlockSource.internal
+    }))
   }
 
-  useEffect(() => setHasMore(true), [searchQuery, setHasMore])
+  const transformedHits = transformItems(hits)
+
+  async function handleFetchMore(): Promise<void> {
+    return await new Promise<void>((resolve) => {
+      showMore()
+      resolve()
+    })
+  }
```

Comment:
Do you have to wrap it in a new promise or can you just do something like...

```suggestion
  async function handleFetchMore(): Promise<void> {
    showMore()
  }
```